### PR TITLE
docs: Update DNSPolicy API reference

### PR DESCRIPTION
**Pull request by mikenairn**

### CODE REVIEW

File: doc/reference/dnspolicy.md [hunk 1]
```diff
@@ -19,45 +19,35 @@
...
+|-------------------|-----------------------------------------------------------------------------------------------------------------------------------|:------------:|---------------------------------------------------------------------------|
+| `targetRef`       | [Gateway API PolicyTargetReference](https://gateway-api.sigs.k8s.io/geps/gep-713/?h=policytargetreference#policy-targetref-api)   |     Yes      | Reference to a Kuberentes resource that the policy attaches to            |
+| `healthCheck`     | [HealthCheckSpec](#healthcheckspec)                                                                                               |      No      | HealthCheck spec                                                          |
+| `loadBalancing`   | [LoadBalancingSpec](#loadbalancingspec)                                                                                           |      No      | LoadBancking Spec                                                         |
```
Comment (philbrookes) [inline doc/reference/dnspolicy.md:26]: LoadBancking?

Comment (mikenairn) [reply doc/reference/dnspolicy.md]: fixed, along with a few other typos

File: doc/reference/dnspolicy.md [hunk 2]
```diff
@@ -19,45 +19,35 @@
...
+|-------------------|-----------------------------------------------------------------------------------------------------------------------------------|:------------:|---------------------------------------------------------------------------|
+| `targetRef`       | [Gateway API PolicyTargetReference](https://gateway-api.sigs.k8s.io/geps/gep-713/?h=policytargetreference#policy-targetref-api)   |     Yes      | Reference to a Kubernetes resource that the policy attaches to            |
+| `healthCheck`     | [HealthCheckSpec](#healthcheckspec)                                                                                               |      No      | HealthCheck spec                                                          |
+| `loadBalancing`   | [LoadBalancingSpec](#loadbalancingspec)                                                                                           |      No      | LoadBalancing Spec                                                        |
```
Comment (pehala) [inline doc/reference/dnspolicy.md:26]: It should be probably be Required as loadbalancing.geo and loadbalancing.weight are required

Comment (maleck13) [reply doc/reference/dnspolicy.md:26]: @pehala can you create an issue for that where it can be discussed

**Review by maleck13 — APPROVED** [review]

File: doc/reference/dnspolicy.md (hunk 2 above)
Comment (pehala) [reply doc/reference/dnspolicy.md:26]: I think it is just a small addition to this doc, loadbalancing section is required when strategy is loadbalanced, I would just mention it here

**Review by pehala — APPROVED** [review]

**Pull request merged**